server/devdocs/bin/generate-proptypes-index.js

```diff
@@ -4,31 +4,99 @@
  * This file generates an index of proptypes by component displayname, slug and folder name
  */
 
+const startTime = process.hrtime();
+
 /**
  * External Dependencies
  */
-
+require( 'babel-register' );
 const fs = require( 'fs' );
 const path = require( 'path' );
 const reactDocgen = require( 'react-docgen' );
+const { getPropertyName, getMemberValuePath, resolveToValue } = require( 'react-docgen/dist/utils' );
+const util = require( 'client/devdocs/docs-example/util' );
 
 const root = path.dirname( path.join( __dirname, '..', '..' ) );
-const pathSwap = new RegExp(path.sep, 'g');
+const pathSwap = new RegExp( path.sep, 'g' );
+const handlers = [ ...reactDocgen.defaultHandlers, commentHandler ];
 
 /**
- * Converts a camel cased string into a slug
- * @param {String} name The camel cased string to slugify
- * @return {String}
+ * Replaces **'s in comment blocks and trims comments
+ * @param {string} str The doc string to clean
+ * @return {string} The clean comment block text
  */
-const camelCaseToSlug = ( name ) => {
-	if ( ! name ) {
-		return name;
+function parseDocblock( str ) {
+	const lines = str.split( '\n' );
+	for ( let i = 0, l = lines.length; i < l; i++ ) {
+		lines[ i ] = lines[ i ].replace( /^\s*\*\s?/, '' );
```

Comment:
Interestingly, I originally wrote this against tests I wrote in `react-docgen` and written there. @ehg and I pulled it out and adapted it for here. The `react-docgen` code base is mostly for-loops and no es6 features. Hence how this code inherited a for-loop. It was just overlooked. Thanks for spotting it :D